torchgeo/datasets/agb_live_woody_density.py

```diff
@@ -105,7 +104,7 @@ def _verify(self) -> None:
         # Check if the user requested to download the dataset
         if not self.download:
             raise RuntimeError(
-                f"Dataset not found in `root={self.paths}` and `download=False`, "
+                f"Dataset not found in `root={str(self.paths)}` and `download=False`, "
```

Comment:
This should automatically be cast to a string during formatting, were you seeing an error?

Reply:
Yes , The error occurs specifically for byte. ``error: On Python 3 '{}'.format(b'abc') produces "b'abc'", not 'abc'; use '{!r}'.format(b'abc') if this is desired behavior when using formatted string literals ``.

Reply:
How do I reproduce this error? It works fine for me:
```pycon
> python3 -Werror
Python 3.11.6 (main, Oct 11 2023, 22:49:14) [Clang 15.0.0 (clang-1500.0.40.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> foo = b"foo"
>>> f"{foo} bar"
"b'foo' bar"
```

Reply:
mypy detects this an error. Here is the code and output
code:
``mypy torchgeo/datasets/agb_live_woody_density.py``
output
``torchgeo/datasets/agb_live_woody_density.py:108: error: If x = b'abc' then f"{x}" or "{}".format(x) produces "b'abc'", not "abc". If this is desired behavior, use f"{x!r}" or "{!r}".format(x). Otherwise, decode the bytes  [str-bytes-safe]
Found 3 errors in 2 files (checked 1 source file)``

Reply:
Can we use `{self.paths!r}` or `{self.paths!s}` instead of `{str(self.paths)}`?

Reply:
Yes, I have implemented this fix.